tests/inference/laplace_approx/test_find_map.py

```diff
@@ -185,6 +185,22 @@ def test_find_MAP(
         assert not hasattr(idata, "unconstrained_posterior")
 
 
+def test_find_map_outside_model_context():
+    """
+    Test that find_MAP can be called outside of a model context.
+    """
+    with pm.Model() as m:
+        mu = pm.Normal("mu", 0, 1)
+        sigma = pm.Exponential("sigma", 1)
+        y_hat = pm.Normal("y_hat", mu=mu, sigma=sigma, observed=np.random.normal(size=10))
+
+    idata = find_MAP(model=m, method="L-BFGS-B", use_grad=True, progressbar=False)
+
+    assert hasattr(idata, "posterior")
+    assert hasattr(idata, "fit")
+    assert hasattr(idata, "optimizer_result")
```

Comment:
Is it relevant / worth checking the presence of any data group? I see above a call like the following within `find_MAP`

```python
    idata = add_data_to_inference_data(
        idata=idata, progressbar=False, model=model, compile_kwargs=compile_kwargs
    )
```

Reply:
haha, well another thing I just saw:

Is it necessary to compute deterministics within `add_data_to_inference_data`? See

https://github.com/pymc-devs/pymc-extras/blob/fd2933f2c787755f03e1cfe106be37e9eb508fb1/pymc_extras/inference/laplace_approx/idata.py#L211-L224

I guess that function is not exposed to the user, but I just wanted to raise that potentially silent side effect.

Reply:
it's not necessary, but that's what pm.sample does right?

Reply:
Ahh I see now why it's included